lib/index.d.ts

```diff
@@ -5,31 +5,31 @@ export type CollectionConfig = {
     /**
      * - One or more glob patterns to match files to a collection
      */
-    pattern: string | string[];
+    pattern?: string | string[] | null;
```

Comment:
The root change I want to correct is the ability for `pattern` to be omitted, such that the `collection` frontmatter array is used.

Proof it's optional: https://github.com/metalsmith/collections/blob/362c20c6814b626850a41f8d245cde81f5af66df/src/index.js#L129

I _think_ the rest of the properties are optional as well, as I see them all given defaults.

Reply:
Ah yeah good catch. I need to check if having the null makes sense. Also, currently developing new major and had to make quite some breaking changes: https://github.com/metalsmith/collections/compare/main...release/2.x.
Still need to wrap my head around some things, like how to make "next" and "previous" references work and how to access "all" vs "own" collections from the metadata